pkg/synchronization/controller.go

```diff
@@ -452,6 +476,33 @@ func (c *Controller) createOrUpdateServiceMetadata(logger *zap.Logger, ns *core_
 	return false, nil
 }
 
+// createOpsGenieSecretSpec creates an OpsGenieIntegrations secret object suitable for Secret creation
+func createOpsGenieSecretSpec(ogInt *opsgenieIntegrationManager.IntegrationsResponse, ns *core_v1.Namespace) (core_v1.Secret, error) {
```

Comment:
I think we should define a separate struct in the `orchestration` package and convert from `opsgenieIntegrationManager.IntegrationsResponse` to it somewhere, similar to how we have defined `ServiceProperties` for `service-metadata` ConfigMap to be passed to autowiring functions. We don't want to couple autowiring context to Opsgenie Integration manager response format.

Reply:
Also I think we agreed to store a team name in one of the keys. This may be useful to see which team the integrations are associated with (in case the team changes in Microscope)